File: basic2/Basic1.go
package main

import "fmt"

func main() {
	var values [3]int
	fmt.Scan(&values[0], &values[1], &values[2])

	var T int
	fmt.Scan(&T)

	result := 0

	for i := 0; i < T; i++ {
		sum := 0
		for j := 0; j < 3; j++ {
			for z := 0; z < len(values); z++ {
				var skill int
				fmt.Scan(&skill)

				sum += skill * values[z]
			}
		}

		if sum > result {
			result = sum
		}
	}

	fmt.Println(result)
}

File: basic2/Basic1.java
package basic2;

import java.util.Scanner;

// 백준 23348

public class Basic1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] values = new int[3];
        for (int i = 0; i < 3; i++) {
            values[i] = sc.nextInt();
        }

        int T = sc.nextInt();

        int result = 0;

        for (int i = 0; i < T; i++) {
            int sum = 0;
            for (int j = 0; j < 3; j++) {
                for (int z = 0; z < values.length; z++) {
                    int skill = sc.nextInt();
                    sum += skill * values[z];
                }
            }

            result = sum > result ? sum : result;
        }

        System.out.println(result);
    }
}
